### ISSUE TITLE
---  getting added to start to the string or generated file

### ISSUE THREAD

**Issue by spa5k** [dtolnay/serde-yaml#266]
Input file - 

node.yaml
```yaml
node:
  version: 18.4.0
packageManagers:
  corepack: true
  yarn:
    version: "3.1"
```

Code - 
```rust
let config_file = current_dir()?.join("node.yaml");
let config = fs::read_to_string(config_file)?;
let node_config: NodeConfig = serde_yaml::from_str(&config)?;

println!("Final config - {:?}", node_config);
let config_str = serde_yaml::to_string(&node_config)?;
println!("CONFIG STRING - {:?}", config_str);

serde_yaml::to_writer(&mut file, &config).unwrap();
```
Output - 
```
Final config - NodeConfig { node: Some(Node { version: "18.4.0" }), package_managers: Some(PackageManagers { corepack: Some(true), yarn: Some(Node { version: "3.1" }) }) }

CONFIG STRING - "---\nnode:\n  version: 18.4.0\npackageManagers:\n  corepack: true\n  yarn:\n    version: \"3.1\"\n"
```

Changes in config file - 

```yaml
---
node:
  version: 18.4.0
packageManagers:
  corepack: true
  yarn:
    version: "3.1"
```
        

Expected - 

The output should not be getting the extra "---"
```
yaml-rust = "0.4"
serde_yaml = "0.8"
```     
        
Rust - 1.63 NIGHTLY

**Comment by spa5k** [comment]
It looks like it's expected thing apparently as i see from the tests, any way to mitigate or ignore it?

**Comment by dtolnay** [comment]
Fixed in 0.9.
